gocd/gocd.go

```diff
@@ -236,47 +250,47 @@ func pipelineGET(gocdURL string, pipelineName string) (pipeline Pipeline, etag s
 }
 
 // Push takes a pipeline from a file and sends it to GoCD
-func Push(gocdURL string, path string, group string) (err error) {
+func Push(server *Server, path string, group string) (err error) {
 	pipeline, err := readPipelineJSONFromFile(path)
 	if err != nil {
 		return
 	}
 
-	etag, err := Exist(gocdURL, pipeline.Name)
+	etag, err := Exist(server, pipeline.Name)
 	if err != nil {
```

Comment:
This might be a good place to add a typed error, since we actually don't need to log anything if the pipeline doesn't exist, as that is an expected state.